apis/Google.Cloud.Firestore/Google.Cloud.Firestore/FieldMask.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright 2018, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+using Google.Api.Gax;
+using Google.Cloud.Firestore.V1Beta1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Cloud.Firestore
+{
+    /// <summary>
+    /// An immutable set of field paths, used to limit the data returned by calls to
+    /// <see cref="FirestoreDb.GetAllSnapshotsAsync(IEnumerable{DocumentReference}, System.Threading.CancellationToken)"/>
+    /// and similar calls.
+    /// </summary>
+    public sealed class FieldMask : IEquatable<FieldMask>
+    {
+        // We store the paths in order for the sake of testability: we can predict the protobuf form easily, and this
+        // provides setwise behavior too.
+        private readonly SortedSet<FieldPath> _fieldPaths;
+
+        private FieldMask(IEnumerable<FieldPath> paths)
+        {
+            _fieldPaths = new SortedSet<FieldPath>(paths);
+        }
+
+        /// <summary>
+        /// Creates a path from the given paths. Each path is treated as a dot-separated sequence of field names.
```

Comment:
Creates a mask.

Reply:
Doh! Thanks. I shouldn't code before coffee. Will fix, rebase, push, wait for green, merge.